packages/tasks/src/model-libraries-snippets.ts

```diff
@@ -326,6 +326,50 @@ export const sklearn = (model: ModelData): string[] => {
 	}
 };
 
+export const stable_audio_tools = (model: ModelData): string[] => [
```

Comment:
This is only one model - but they support other models (although there are no pre-trained models supported)


Let me see if I can reduce the snippet size.

Reply:
Here's the shortened snippet.

```python
import torch
import torchaudio
from einops import rearrange
from stable_audio_tools import get_pretrained_model
from stable_audio_tools.inference.generation import generate_diffusion_cond

device = "cuda" if torch.cuda.is_available() else "cpu"

# Download model
model, model_config = get_pretrained_model("stabilityai/stable-audio-open-1.0")
sample_rate = model_config["sample_rate"]
sample_size = model_config["sample_size"]

model = model.to(device)

# Set up text and timing conditioning
conditioning = [{
    "prompt": "128 BPM tech house drum loop",
}]

# Generate stereo audio
output = generate_diffusion_cond(
    model,
    conditioning=conditioning,
    sample_size=sample_size,
    device=device
)

# Rearrange audio batch to a single sequence
output = rearrange(output, "b d n -> d (b n)")

# Peak normalize, clip, convert to int16, and save to file
output = output.to(torch.float32).div(torch.max(torch.abs(output))).clamp(-1, 1).mul(32767).to(torch.int16).cpu()
torchaudio.save("output.wav", output, sample_rate)
```

Reply:
This is too complex imo. Usually the snippets don't have things such as loading into GPU for example. We usually also don't show the actual inference (and just do the loading), but it might be fine for this case. 

Although it's just one model, the library is fine-tuning, so we need to make sure our snippet would work for those.

Reply:
I updated the code snippet to the minified code snippet as mentioned above!

Reply:
(Note: This would generalise to all stable-audio-tools fine-tune for stable-audio-open too)

Reply:
Yes, for comparison, the snippet for stable diffusion looks just like this:

```
from diffusers import DiffusionPipeline

pipeline = DiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
```

So no inference or inputs shown at all. If we want to be consistent we should do something similar. If we think it's useful to provide complete standalone references (which could include inference or fine-tuning tasks), then maybe that's something to consider along the line. The model card for this particular model already includes a complete inference snippet, do you think the same code would still work for any fine-tuned version, that perhaps may be conditioned on different signals (not sure if possible, tbh)?

Reply:
Currently, every fine-tuned model based on the ft code would use the same snippet. The only change would be to init but at that point it should be reported in the model card as that would be specific to th efine-tune. 

This is still new so happy to revisit it later based on how the library evolves. A complete reproducible snippet goes a long way to adoption. 

Just instantiating the pipeline would only lead to so much. I'm sure there were reasons for it that I'm not aware of haha.

Reply:
That's a good point, happy to hear other opinions :)

Reply:
no preference on my side!

Reply:
No strong opinion either especially since it can be reassessed later on if we see it brakes on some finetunes

Reply:
Cool! Should we merge with the current snippet for now? I'm setting a reminder to revisit this in 20 or so days.

Reply:
Fine for me!